cores/rp2040/SerialUSB.cpp

```diff
@@ -174,8 +175,19 @@ SerialUSB::operator bool() {
 static bool _dtr = false;
 static bool _rts = false;
 static int _bps = 115200;
+static volatile bool _rebooting = false;
 static void CheckSerialReset() {
-    if ((_bps == 1200) && (!_dtr)) {
+    if (!_rebooting && (_bps == 1200) && (!_dtr)) {
+        _rebooting = true;
+        // execute any pending tasks, so the host is happy
+        for(int i=0; i<1000; i++) tud_task();
+        // disable nvic irq, so that we don't get bothered anymore
+        irq_set_enabled(USBCTRL_IRQ, false);
+        // reset the whole USB hardware block
+        reset_block(RESETS_RESET_USBCTRL_BITS);
+        unreset_block(RESETS_RESET_USBCTRL_BITS);
```

Comment:
We'll reset the USB block, pull it out if reset (I assume all IOs floating?), and then it'll go back into reset 3ms later because the WDT will reset the whole chip.  Is the `unreset` here really doing anything in that case?

Reply:
If you want to float the pins, then wouldn't it be better to just leave it in reset anyway?  The delay(3ms) stays, sure, to guarantee some "disconnect" time for the host to recognize things are amiss.  I just would have thought the reset state of the block wouldn't drive anything anyway so there's no need to pull out of reset...

Reply:
Since I'm unfamiliar with the silicon, or it's eratta, it seems best to be safe, rather than sorry.
I've seen stranger things happen, where you reset a block that's resetting, and it triggers a hard fault, since it is already in reset. No idea if this is implemented, or not. It's always best to leave hardware in a known state, even if the WDT does a hardware reset. 
There could be possible corner cases/side effects, especially since there are two cores involved.
I will leave it in, but, if it proves to not be a problem, then it is probably fine to remove it.